### PR TITLE
change kit on network switch and add feeCurrency to the provider

### PR DESCRIPTION
**Pull request by Bijan-Massoumi**
- fix a bug where changing network in metamask doesn't refresh the contractKit object
- add configurable feeCurrency

### CODE REVIEW

File: packages/use-contractkit/src/screens/cew.tsx [hunk 1]
```diff
@@ -8,10 +8,15 @@ import { ConnectorProps } from '.';
...
+    initError: error,
+    feeCurrency,
+  } = useContractKitInternal();
 
   const initialiseConnection = useCallback(async () => {
-    const connector = new CeloExtensionWalletConnector(network);
+    const connector = new CeloExtensionWalletConnector(network, feeCurrency);
     await initConnector(connector);
     void onSubmit(connector);
   }, [initConnector, network, onSubmit]);
```
Comment (nicolasbrugneaux) [inline packages/use-contractkit/src/screens/cew.tsx:22]: probably good to remove the lint warning here by adding the `feeCurrency` to the depsArray

File: packages/use-contractkit/src/use-contract-kit-methods.ts [hunk 2]
```diff
@@ -42,7 +48,17 @@ export function useContractKitMethods(
...
+
+          if (network) {
+            dispatch('setNetwork', network);
+            initialisedConnector.updateKitWithNetwork &&
+              initialisedConnector
+                .updateKitWithNetwork(network)
+                .then(() => {
+                  dispatch('initialisedConnector', initialisedConnector);
+                })
+                .catch((e) => console.log(e));
```
Comment (nicolasbrugneaux) [inline packages/use-contractkit/src/use-contract-kit-methods.ts:60]: do we really just wanna ignore errors here?

Comment (Bijan-Massoumi) [reply packages/use-contractkit/src/use-contract-kit-methods.ts:60]: Good catch

File: packages/use-contractkit/src/contract-kit-provider.tsx [hunk 3]
```diff
@@ -92,6 +94,7 @@ export const ContractKitProvider: React.FC<ContractKitProviderProps> = ({
   dapp,
   network = Mainnet,
   networks = DEFAULT_NETWORKS,
+  feeCurrency = CeloContract.StableToken,
```
Comment (dckesler) [inline packages/use-contractkit/src/contract-kit-provider.tsx:97]: Should this also be `GoldToken`?

File: packages/use-contractkit/src/use-contract-kit-methods.ts [hunk 4]
```diff
@@ -116,6 +146,14 @@ export function useContractKitMethods(
     return initialisedConnection.kit;
   }, [connect, connector, initConnector]);
 
+  const updateFeeCurrency = useCallback(
```
Comment (dckesler) [inline packages/use-contractkit/src/use-contract-kit-methods.ts:149]: We should add this to the return object.

File: packages/use-contractkit/src/connectors/useMetaMaskConnector.ts [hunk 5]
```diff
@@ -17,7 +18,7 @@ export function useMetaMaskConnector(
   useEffect(() => {
     let stale;
     void (async () => {
-      const connector = new MetaMaskConnector(network);
+      const connector = new MetaMaskConnector(network, feeCurrency);
```
Comment (dckesler) [inline packages/use-contractkit/src/connectors/useMetaMaskConnector.ts:21]: `feeCurrency` should be added to the deps array too

**Review by dckesler — CHANGES_REQUESTED** [review]
Overall really good a few small comments. On top of that I have two requests for additional content.
1. I think since we're adding a new method to the API that we should add something to the README about `updateFeeCurrency`.
2. I don't feel too strongly about this but it could be nice to add `feeCurrency` to our local storage system as well. Just so that whatever feeCurrency they picked sticks around.

**Review by dckesler — APPROVED** [review]

**Pull request merged**